text/0005-application-formalization.md

```diff
@@ -0,0 +1,194 @@
+* Start Date: 10/09/2023
+* Contributors:
+  * [Craig McNally](cmcnally@ebsco.com)
+  * [Vince Bareau](vbareau@ebsco.com)
+* RFC PRs:
+  * PRELIMINARY REVIEW: https://github.com/folio-org/rfcs/pull/14
+  * DRAFT REFINEMENT: https://github.com/folio-org/rfcs/pull/22
+  * PUBLIC REVIEW: TBD
+  * FINAL REVIEW: TBD
+* Outcome: <!-- Leave this blank. Will eventually be either ACCEPTED or REJECTED -->
+
+# Application Formalization
+
+## Summary
+It is proposed that an Application be defined as the minimal but complete set of elements which together are intended to deliver a specific solution to Folio. The intention is to group together all the closely related components which will need to evolve together, but independently of other parts (i.e. other applications) of Folio.
+
+## Motivation
+The existing Folio project lacks a formal definition of an Application structure. A consequence is that Folio is stuck in a mode of monolithic releases which are ever increasing in size.  There have also been repeated complaints that the setup and management of Folio can be difficult due to the number modules involved.
+
+## Scope
+* Application composition
+* Application versioning
+* Application dependencies
+* Application definition/descriptors
+* Backward compatibility with Folio instances not adopting Applications
+* Application management is ***out of scope***
+* Application Stores and Marketplaces are ***out of scope***
+* Application-specific UI bundles is ***out of scope***
+* How movement of modules between Applications will work is ***out of scope***
+* How bounded contexts and cross-module database access could be introduced is ***out of scope***
+
+## Detailed Explanation/Design
+
+### Terminology
+* Application:  The minimal but complete set of elements which together are intended to deliver a specific solution to Folio.
+* Bounded context:  The boundary of the microservice.  It "owns" the storage for the service.
```

Comment:
Is it worth going further with the definition of "Bounded context"? Discussion have suggested that there's an impression the storage determines the boundary rather than being determined by the boundary.  Maybe something like "that boundary is determined by the processes being modeled by the microservice, the bounded context includes the code and data artifacts that implement that model". Hopefully that makes it more clear why it "owns" the storage for the service.

Reply:
I'd prefer not to get too far into the bounded context details here given that topic will be covered in a separate RFC

Reply:
@marcjohnson-kint , @starsplatter , @inkuss what do you think?  Is the definition as it stands right now clear enough, or does it need to be clarified?

Reply:
> what do you think? Is the definition as it stands right now clear enough, or does it need to be clarified?

Apologies, I don't think I can give useful feedback on this due to how my understanding of bounded contexts is too different

Reply:
In the interest of moving this forward I'm going to mark this as resolved.  We can revisit this during PUBLIC REVIEW if needed.